eventloop: TestHandler is flaky; possible deadlock

```console
$ go test -run TestHandler -count=20000
--- FAIL: TestHandler (1.00s)
    eventloop_test.go:31: timed out
--- FAIL: TestHandler (1.00s)
    eventloop_test.go:31: timed out
--- FAIL: TestHandler (1.00s)
    eventloop_test.go:31: timed out
--- FAIL: TestHandler (1.00s)
    eventloop_test.go:31: timed out
FAIL
exit status 1
FAIL	github.com/relab/hotstuff/eventloop	4.776s
```

Sometimes tests pass just fine (and finishes quickly), but running many times you should be able to reproduce. It does not help to increase the context's timeout, so I suspect this is a deadlock because non-failing test executions finish fast.

I discovered this when upgrading the dependencies in PR #118, which itself doesn't touch the event loop. However, it reproduces also in master per commit `6c1fcb7b4b413`.

--- a/.github/codecov.yml
+++ b/.github/codecov.yml
@@ -1,0 +1,6 @@
+coverage:
+  status:
+    project:
+      default:
+        target: auto
+        threshold: 1%

--- a/eventloop/eventloop_test.go
+++ b/eventloop/eventloop_test.go
@@ -22,6 +22,9 @@ func TestHandler(t *testing.T) {
 	defer cancel()
 	go el.Run(ctx)
 
+	// wait for the event loop to start
+	time.Sleep(1 * time.Millisecond)
+
 	want := testEvent(42)
 	el.AddEvent(want)
 

